lib/src/otx/data/samplers/balanced_sampler.py

```diff
@@ -61,7 +64,11 @@ def __init__(
         super().__init__(dataset)
 
         # img_indices: dict[label: list[idx]]
-        ann_stats = get_idx_list_per_classes(dataset.dm_subset)
+        if isinstance(dataset.dm_subset, DatasetSubset):
+            ann_stats = get_idx_list_per_classes(dataset.dm_subset)
```

Comment:
If possible, I would suggest to add a `get_idx_list_per_classes()` method to the old `OTXDataset`. That way, we don’t need to differentiate here between OTXDataset and OTXDatasetNew.

Reply:
Good point, fixed.